File: 1436-destination-city/1436-destination-city.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
var destCity = function(paths) {
    const cityMap = new Map();
    for(const [city, des] of paths) {
        cityMap.set(city, des);
    }
    for(const [city, des] of cityMap) {
        if(!cityMap.has(des)) return des;
    }
};

File: 1436-destination-city/1436-destination-city.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        city_dict = {}
        for city_a, city_b in paths:
            city_dict[city_a] = city_b
        
        for value in city_dict.values():
            if value not in city_dict:
                return value
                